ext/mysql2/client.c

```diff
@@ -271,13 +264,10 @@ static VALUE invalidate_fd(int clientfd)
 static void *nogvl_close(void *ptr) {
   mysql_client_wrapper *wrapper = ptr;
 
-  if (wrapper->client) {
-    mysql_close(wrapper->client);
-    xfree(wrapper->client);
-    wrapper->client = NULL;
-    wrapper->connected = 0;
-    wrapper->active_thread = Qnil;
-  }
+  mysql_close(wrapper->client);
```

Comment:
I need to look back to where this defensive programming was added, but we definitely had cases of invalid pointer dereferencing in the past that this check was intended to prevent.